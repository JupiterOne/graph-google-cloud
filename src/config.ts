import {
  IntegrationExecutionContext,
  IntegrationLogger,
} from '@jupiterone/integration-sdk-core';
import { ServiceUsageClient } from './steps/service-usage/client';
import { IntegrationConfig, SerializedIntegrationConfig } from './types';
import { deserializeIntegrationConfig } from './utils/integrationConfig';
import { google } from 'googleapis';
import { handleApiClientError } from './google-cloud/client';
import { ServiceUsageListFilter } from './google-cloud/types';

export async function executeTestRequest(
  config: IntegrationConfig,
  logger: IntegrationLogger,
) {
  try {
    const client = google.serviceusage({ version: 'v1', retry: false });
    const googleClient = new ServiceUsageClient({ config }, logger);
    const auth = await googleClient.getAuthenticatedServiceClient();

    await client.services.list({
      parent: `projects/${config.serviceAccountKeyConfig.project_id}`,
      pageSize: 200,
      auth: auth,
      filter: ServiceUsageListFilter.ENABLED,
    });
  } catch (err) {
    throw handleApiClientError(err);
  }
}

export async function validateInvocation(
  context: IntegrationExecutionContext<SerializedIntegrationConfig>,
) {
  const { instance, logger } = context;
  const { config: serializedIntegrationConfig } = instance;
  const config = (context.instance.config = deserializeIntegrationConfig(
    serializedIntegrationConfig,
  ));

  await executeTestRequest(config, logger);
}
