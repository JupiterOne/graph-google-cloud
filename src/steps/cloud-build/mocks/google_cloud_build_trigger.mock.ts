export default {
  id: '0f5d9ec8-f7d7-4067-8912-baf2133be725',
  createTime: '2022-08-24T22:09:57.281610382Z',
  name: 'testing',
  gitFileSource: {
    path: 'cloudbuild.yaml',
    uri: 'https://github.com/sample-user/sample-repo',
    repoType: 'GITHUB',
    revision: 'refs/heads/master',
  },
  sourceToBuild: {
    uri: 'https://github.com/sample-user/sample-repo',
    ref: 'refs/heads/master',
    repoType: 'GITHUB',
  },
};
