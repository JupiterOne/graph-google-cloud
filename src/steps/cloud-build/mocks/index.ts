import google_cloud_build from './google_cloud_build.mock';
import google_cloud_build_trigger from './google_cloud_build_trigger.mock';

export default { google_cloud_build, google_cloud_build_trigger };
