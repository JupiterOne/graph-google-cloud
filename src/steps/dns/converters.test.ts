import { getMockDNSManagedZone, getMockDNSPolicy } from '../../../test/mocks';
import {
  createDNSManagedZoneEntity,
  createDNSPolicyEntity,
} from './converters';
import { DEFAULT_INTEGRATION_CONFIG_PROJECT_ID } from '../../../test/config';

describe('#createDNSManagedZoneEntity', () => {
  test('should convert to entity', () => {
    expect(
      createDNSManagedZoneEntity(getMockDNSManagedZone()),
    ).toMatchSnapshot();
  });

  test('should set dnssec to "on" when dnssec is enabled', () => {
    expect(
      createDNSManagedZoneEntity(
        getMockDNSManagedZone({
          cloudLoggingConfig: {
            enableLogging: true,
          },
          dnssecConfig: {
            state: 'on',
            nonExistence: 'nsec3',
            kind: 'dns#managedZoneDnsSecConfig',
          },
        }),
      ),
    ).toMatchSnapshot();
  });

  test('should set value to keySigningAlgorithm when it is present', () => {
    expect(
      createDNSManagedZoneEntity(
        getMockDNSManagedZone({
          cloudLoggingConfig: {
            enableLogging: null,
          },
          dnssecConfig: {
            state: 'on',
            defaultKeySpecs: [
              {
                keyType: 'keySigning',
                algorithm: 'rsasha256',
                keyLength: 2048,
                kind: 'dns#dnsKeySpec',
              },
            ],
            nonExistence: 'nsec3',
            kind: 'dns#managedZoneDnsSecConfig',
          },
        }),
      ),
    ).toMatchSnapshot();
  });

  test('should set value to zoneSigningAlgorithm when it is present', () => {
    expect(
      createDNSManagedZoneEntity(
        getMockDNSManagedZone({
          cloudLoggingConfig: {
            enableLogging: false,
          },
          dnssecConfig: {
            state: 'on',
            defaultKeySpecs: [
              {
                keyType: 'zoneSigning',
                algorithm: 'rsasha256',
                keyLength: 1024,
                kind: 'dns#dnsKeySpec',
              },
            ],
            nonExistence: 'nsec3',
            kind: 'dns#managedZoneDnsSecConfig',
          },
        }),
      ),
    ).toMatchSnapshot();
  });
});

describe('#createDNSPolicyEntity', () => {
  test('should convert to entity', () => {
    expect(
      createDNSPolicyEntity(
        getMockDNSPolicy(),
        DEFAULT_INTEGRATION_CONFIG_PROJECT_ID,
      ),
    ).toMatchSnapshot();
  });
});
