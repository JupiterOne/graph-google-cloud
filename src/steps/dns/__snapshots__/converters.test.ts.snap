// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createDNSManagedZoneEntity should convert to entity 1`] = `
{
  "_class": [
    "DomainZone",
  ],
  "_key": "778386319953398157",
  "_rawData": [
    {
      "name": "default",
      "rawData": {
        "creationTime": "2021-01-22T16:48:27.272Z",
        "description": "",
        "dnsName": "example.com.",
        "id": "778386319953398157",
        "kind": "dns#managedZone",
        "name": "example-zone",
        "nameServers": [
          "ns-cloud-a1.googledomains.com.",
          "ns-cloud-a2.googledomains.com.",
          "ns-cloud-a3.googledomains.com.",
          "ns-cloud-a4.googledomains.com.",
        ],
        "visibility": "public",
      },
    },
  ],
  "_type": "google_dns_managed_zone",
  "cloudLoggingStatus": "disabled",
  "createdOn": 1611334107272,
  "description": "",
  "displayName": "example-zone",
  "dnssec": undefined,
  "domainName": "example.com.",
  "id": "778386319953398157",
  "keySigningAlgorithm": undefined,
  "kind": "dns#managedZone",
  "name": "example-zone",
  "visibility": "public",
  "zoneSigningAlgorithm": undefined,
}
`;

exports[`#createDNSManagedZoneEntity should set dnssec to "on" when dnssec is enabled 1`] = `
{
  "_class": [
    "DomainZone",
  ],
  "_key": "778386319953398157",
  "_rawData": [
    {
      "name": "default",
      "rawData": {
        "cloudLoggingConfig": {
          "enableLogging": true,
        },
        "creationTime": "2021-01-22T16:48:27.272Z",
        "description": "",
        "dnsName": "example.com.",
        "dnssecConfig": {
          "kind": "dns#managedZoneDnsSecConfig",
          "nonExistence": "nsec3",
          "state": "on",
        },
        "id": "778386319953398157",
        "kind": "dns#managedZone",
        "name": "example-zone",
        "nameServers": [
          "ns-cloud-a1.googledomains.com.",
          "ns-cloud-a2.googledomains.com.",
          "ns-cloud-a3.googledomains.com.",
          "ns-cloud-a4.googledomains.com.",
        ],
        "visibility": "public",
      },
    },
  ],
  "_type": "google_dns_managed_zone",
  "cloudLoggingStatus": "enabled",
  "createdOn": 1611334107272,
  "description": "",
  "displayName": "example-zone",
  "dnssec": "on",
  "domainName": "example.com.",
  "id": "778386319953398157",
  "keySigningAlgorithm": undefined,
  "kind": "dns#managedZone",
  "name": "example-zone",
  "visibility": "public",
  "zoneSigningAlgorithm": undefined,
}
`;

exports[`#createDNSManagedZoneEntity should set value to keySigningAlgorithm when it is present 1`] = `
{
  "_class": [
    "DomainZone",
  ],
  "_key": "778386319953398157",
  "_rawData": [
    {
      "name": "default",
      "rawData": {
        "cloudLoggingConfig": {
          "enableLogging": null,
        },
        "creationTime": "2021-01-22T16:48:27.272Z",
        "description": "",
        "dnsName": "example.com.",
        "dnssecConfig": {
          "defaultKeySpecs": [
            {
              "algorithm": "rsasha256",
              "keyLength": 2048,
              "keyType": "keySigning",
              "kind": "dns#dnsKeySpec",
            },
          ],
          "kind": "dns#managedZoneDnsSecConfig",
          "nonExistence": "nsec3",
          "state": "on",
        },
        "id": "778386319953398157",
        "kind": "dns#managedZone",
        "name": "example-zone",
        "nameServers": [
          "ns-cloud-a1.googledomains.com.",
          "ns-cloud-a2.googledomains.com.",
          "ns-cloud-a3.googledomains.com.",
          "ns-cloud-a4.googledomains.com.",
        ],
        "visibility": "public",
      },
    },
  ],
  "_type": "google_dns_managed_zone",
  "cloudLoggingStatus": "disabled",
  "createdOn": 1611334107272,
  "description": "",
  "displayName": "example-zone",
  "dnssec": "on",
  "domainName": "example.com.",
  "id": "778386319953398157",
  "keySigningAlgorithm": "rsasha256",
  "kind": "dns#managedZone",
  "name": "example-zone",
  "visibility": "public",
  "zoneSigningAlgorithm": undefined,
}
`;

exports[`#createDNSManagedZoneEntity should set value to zoneSigningAlgorithm when it is present 1`] = `
{
  "_class": [
    "DomainZone",
  ],
  "_key": "778386319953398157",
  "_rawData": [
    {
      "name": "default",
      "rawData": {
        "cloudLoggingConfig": {
          "enableLogging": false,
        },
        "creationTime": "2021-01-22T16:48:27.272Z",
        "description": "",
        "dnsName": "example.com.",
        "dnssecConfig": {
          "defaultKeySpecs": [
            {
              "algorithm": "rsasha256",
              "keyLength": 1024,
              "keyType": "zoneSigning",
              "kind": "dns#dnsKeySpec",
            },
          ],
          "kind": "dns#managedZoneDnsSecConfig",
          "nonExistence": "nsec3",
          "state": "on",
        },
        "id": "778386319953398157",
        "kind": "dns#managedZone",
        "name": "example-zone",
        "nameServers": [
          "ns-cloud-a1.googledomains.com.",
          "ns-cloud-a2.googledomains.com.",
          "ns-cloud-a3.googledomains.com.",
          "ns-cloud-a4.googledomains.com.",
        ],
        "visibility": "public",
      },
    },
  ],
  "_type": "google_dns_managed_zone",
  "cloudLoggingStatus": "disabled",
  "createdOn": 1611334107272,
  "description": "",
  "displayName": "example-zone",
  "dnssec": "on",
  "domainName": "example.com.",
  "id": "778386319953398157",
  "keySigningAlgorithm": undefined,
  "kind": "dns#managedZone",
  "name": "example-zone",
  "visibility": "public",
  "zoneSigningAlgorithm": "rsasha256",
}
`;

exports[`#createDNSPolicyEntity should convert to entity 1`] = `
{
  "_class": [
    "Ruleset",
  ],
  "_key": "projects/j1-gc-integration-dev-v2/policies/2869664235641544935",
  "_rawData": [
    {
      "name": "default",
      "rawData": {
        "description": "",
        "enableInboundForwarding": true,
        "enableLogging": true,
        "id": "2869664235641544935",
        "kind": "dns#policy",
        "name": "test-policy",
        "networks": [
          {
            "kind": "dns#policyNetwork",
            "networkUrl": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/networks/public-compute-app-vpc",
          },
          {
            "kind": "dns#policyNetwork",
            "networkUrl": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/networks/default",
          },
        ],
      },
    },
  ],
  "_type": "google_dns_policy",
  "createdOn": undefined,
  "description": "",
  "displayName": "test-policy",
  "enableInboundForwarding": true,
  "enableLogging": true,
  "id": "2869664235641544935",
  "kind": "dns#policy",
  "name": "test-policy",
  "networks": [
    "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/networks/public-compute-app-vpc",
    "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/networks/default",
  ],
}
`;
